src/lib/pages/backoffice/Acquisition/Order/OrderDetails/OrderDocumentRequest.js

```diff
@@ -0,0 +1,93 @@
+import { PropTypes } from 'prop-types';
+import React from 'react';
+import { withCancel } from '@api/utils';
+import { documentRequestApi } from '@api/documentRequests/documentRequest';
+import { Link } from 'react-router-dom';
+import { BackOfficeRoutes } from '@routes/urls';
+import { Message } from 'semantic-ui-react';
+import { Loader } from '@components/Loader';
+import _isEmpty from 'lodash/isEmpty';
+
+export class OrderDocumentRequest extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cancellableLiterature: {},
+      isLoading: true,
+      hasError: false,
+    };
+  }
+
+  componentDidMount() {
+    const {
+      orderId: { pid },
+    } = this.props;
+    this.fetchDocumentRequest(pid);
+  }
+
+  componentWillUnmount() {
+    this.cancellableDocumentRequest && this.cancellableDocumentRequest.cancel();
+  }
+
+  fetchDocumentRequest = async (pid) => {
+    try {
+      this.cancellableDocumentRequest = withCancel(
+        documentRequestApi.list(
+          documentRequestApi.query().withPhysicalItemProviderPid(pid).qs()
+        )
+      );
+      const response = await this.cancellableDocumentRequest.promise;
```

Comment:
you could avoid the check for the `isEmpty` below:

```
this.setState({
        cancellableLiterature: response.data.hits.length ? response.data.hits[0] : null,
        hasError: false,
        isLoading: false,
      });
```

and then below in the `renderResultsOrEmpty` method:
```
renderResultsOrEmpty(documentRequest) {
    return documentRequest ? (
        <Link to={BackOfficeRoutes.documentRequestDetailsFor(documentRequest.metadata.pid)}>
```
it is a very small improvement that will avoid checking the `empty` on each re-render of the component ;)